clients/src/main/java/org/apache/kafka/common/metrics/stats/Frequencies.java

```diff
@@ -0,0 +1,192 @@
+ * object is a {@link CompoundStat}, and so it can be {@link org.apache.kafka.common.metrics.Sensor#add(CompoundStat)
+ * added directly to a Sensor} so the metrics are created automatically.
+ */
+public class Frequencies extends SampledStat implements CompoundStat {
+
+    /**
+     * Create a Frequencies instance with metrics for the frequency of a boolean sensor that records 0.0 for
+     * false and 1.0 for true.
+     *
+     * @param falseMetricName the name of the metric capturing the frequency of failures; may be null if not needed
+     * @param trueMetricName  the name of the metric capturing the frequency of successes; may be null if not needed
+     * @return the Frequencies instance; never null
+     * @throws IllegalArgumentException if both {@code falseMetricName} and {@code trueMetricName} are null
+     */
+    public static Frequencies forBooleanValues(MetricName falseMetricName, MetricName trueMetricName) {
+        List<Frequency> frequencies = new ArrayList<>();
+        if (falseMetricName != null) {
+            frequencies.add(new Frequency(falseMetricName, 0.0));
+        }
+        if (trueMetricName != null) {
+            frequencies.add(new Frequency(trueMetricName, 1.0));
+        }
+        if (frequencies.isEmpty()) {
+            throw new IllegalArgumentException("Must specify at least one metric name");
+        }
+        Frequency[] frequencyArray = frequencies.toArray(new Frequency[frequencies.size()]);
+        return new Frequencies(2, 0.0, 1.0, frequencyArray);
+    }
+
+    private final Frequency[] frequencies;
+    private final BinScheme binScheme;
+
+    /**
+     * Create a Frequencies that captures the values in the specified range into the given number of buckets,
+     * where the buckets are centered around the minimum, maximum, and intermediate values.
+     *
+     * @param buckets     the number of buckets
+     * @param min         the minimum value to be captured
+     * @param max         the maximum value to be captured
+     * @param frequencies the list of {@link Frequency} metrics, which at most should be one per bucket centered
+     *                    on the bucket's value, though not every bucket need to correspond to a metric if the
+     *                    value is not needed
+     * @throws IllegalArgumentException if any of the {@link Frequency} objects do not have a
+     *                                  {@link Frequency#centerValue() center value} within the specified range
+     */
+    public Frequencies(int buckets, double min, double max, Frequency... frequencies) {
+        super(0.0);
+        if (max < min) {
+            throw new IllegalArgumentException("The maximum value " + max
+                                                       + " must be greater than the minimum value " + min);
+        }
+        if (buckets < 1) {
+            throw new IllegalArgumentException("Must be at least 2 buckets");
```

Comment:
The message doesn't seem to match the condition above.

Reply:
Good catch. Will fix shortly.